src/main/java/Nehsik.java

```diff
@@ -0,0 +1,93 @@
+import java.util.Scanner;
+public class Nehsik {
+    public static final int MAX_TASKS = 100;
+
+    public static void main(String[] args) {
+        displayGreetings();
+        
+        Scanner in = new Scanner(System.in);
+        Task[] taskList = new Task[MAX_TASKS];
+        int taskIndex = 0;
+
+        while (true) {
+            String command = in.nextLine();
+            if (command.equals("list")) {
+                printLine();
+                System.out.println("Here are the tasks in your list:");
+                for (int i = 0; i < taskIndex; i++) {
+                    System.out.println((i + 1) + "." + taskList[i].toString());
+                }
+                printLine();
+            } else if (command.equals("bye")) {
+                break;
+            } else if (command.startsWith("mark")){
+                printLine();
+                System.out.println("Nice! I've marked this task as done:");
+                int taskNum = Integer.parseInt(command.substring(5)) - 1;
+                taskList[taskNum].markAsDone();
+                System.out.println(taskList[taskNum].toString());
+                printLine();
+            } else if (command.startsWith("unmark")) {
+                printLine();
+                System.out.println("OK, I've marked this task as not done yet:");
+                int taskNum = Integer.parseInt(command.substring(7)) - 1;
+                taskList[taskNum].markAsUndone();
+                System.out.println(taskList[taskNum].toString());
+                printLine();
+            }  else if (command.startsWith("todo")) {
+                taskList[taskIndex] = new Todo(command);
+
+                printLine();
+                System.out.println("Got it. I've added this task:");
```

Comment:
Minimise Code Duplication! It can be seen in line 41-45 and line 52-56 as well as line 52-56 and line 64-68.